### ISSUE TITLE
Provide a "pipe-able" spack find output format.

### ISSUE THREAD

**Issue by hartzell** [LLNL/spack#1369]
I'd like to be able to pipe the output of `spack find ...` into other commands, but the colorization gets in the way, as do the compiler headers.

My naive use case was to play with `spack view`, linking all of the things that I've built with _%gcc@5.4.0_ but not _%gcc@4.8.5_ and only one of the several perls I have installed and `grep -v ...` a couple of things.
### colorization

I thought perhaps short output might not be colored or .... but it seems like `color=True` is hardwired in _lib/spack/spack/cmd/__init__.py_.

[This](http://www.commandlinefu.com/commands/view/3584/remove-color-codes-special-characters-with-sed) (thanks Google!) works but is not very appealing

```
spack find | sed -r "s/\x1B\[([0-9]{1,2}(;[0-9]{1,2})?)?[m|K]//g"
```

I could imagine adding a --no_color option to the find subcommand, or teaching the find subcommand to not colorize the output if it's writing to a pipe (not writing to a tty).

Would that be generally useful?
### compiler headers

Rather than have a header separating the packages by compiler, it seems like would be useful to have a way to just include them in the spec: `tcl@8.6.5%gcc@5.4.0`, which is what is useful for e.g. `spec view add`.
### Useful?

Perhaps if it detects that it's writing to a pipe (not a tty) it could drop the headers and produce a single uncolored line with compiler info as part of the spec.  Options that produce multi word output (e.g. `-l`) would continue to (people can `perl -lane ...` it or whatever strikes their fancy).

If this seems useful and someone will mentor my Python, I'm happy to take a swing at it.

**Comment by adamjstewart** [comment]
@citibeth might be interested in this. See #656

**Comment by adamjstewart** [comment]
I think this will be fixed by #3013, which allows you to run `spack --no-color find`.

**Comment by tgamblin** [comment]
@hartzell: does piping `spack find` work for you now?
